src/main/java/scm/address/logic/commands/ImportCommand.java

```diff
@@ -119,19 +143,63 @@ public void retrievePersonsFromFile(List<JsonAdaptedPerson> savedPersons)
      * @return A List of JsonAdaptedPerson present inside the {@code file}.
      * @throws DataLoadingException If the {@code file} is unable to be loaded.
      */
-    public List<JsonAdaptedPerson> readPersons(File file)
+    private List<JsonAdaptedPerson> readPersons(File file)
             throws DataLoadingException {
         JsonAddressBookStorage curStorage = new JsonAddressBookStorage(file.toPath());
         Optional<ReadOnlyAddressBook> readOnlyAddressBook = curStorage.readAddressBook();
         return readOnlyAddressBook
                 .orElseThrow(() ->
-                        new DataLoadingException(new Exception("Cannot load file: " + file.getPath())))
+                        new DataLoadingException(new Exception(MESSAGE_FILE_LOADING_ERROR + file.getPath())))
                 .getPersonList()
                 .stream()
                 .map(JsonAdaptedPerson::new)
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Reads the csv data inside {@code file} and returns a List of JsonAdaptedPersons.
+     *
+     * @param file A csv file.
+     * @return A List of JsonAdaptedPerson present inside the {@code file}.
+     * @throws DataLoadingException If the {@code file} is unable to be loaded.
+     */
+    private List<JsonAdaptedPerson> readPersonsFromCsv(File file) throws DataLoadingException {
+        String filePath = file.getPath();
+        List<JsonAdaptedPerson> persons = new ArrayList<>();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(filePath));
+            String line = "";
+            String splitBy = ",(?=([^\"]*\"[^\"]*\")*[^\"]*$)";
+            String headers = br.readLine();
+            while ((line = br.readLine()) != null) {
```

Comment:
This could perhaps be abstracted into different methods.

Reply:
I extracted out a function, however, it doesn't help much, as the code chuck needs to be in one chunk. It does, however, decrease nesting by one depth. Thanks!